### DISCUSSION TITLE
Нужна помощь.

### DISCUSSION THREAD

**Discussion by wezzid** [MrZillaGold/VK2Discord#404] [Q&A] [answered]
Пытаюсь запустить скрипт на Heroku, чтобы работал автоматически без ПК. Выдает ошибку, нашел решение вот тут: https://ru.stackoverflow.com/questions/1307095/%D0%9E%D1%88%D0%B8%D0%B1%D0%BA%D0%B0-%D0%B2%D0%BE-%D0%B2%D1%80%D0%B5%D0%BC%D1%8F-deploy-heroku , но у меня скрипт уже отличается от того, что было раньше. Прикрепляю скрин лога, как раз на этом скрипте старт он начинает запинаться, пробовал удалять полностью, ошибка все равно возникает
![image](https://user-images.githubusercontent.com/97441670/148793257-485c8913-3ff9-4fbb-b113-4a0a552d6415.png)

2022-01-10T15:29:07.898365+00:00 heroku[worker.1]: Starting process with command npm start
2022-01-10T15:29:08.528280+00:00 heroku[worker.1]: State changed from starting to up
2022-01-10T15:29:08.971921+00:00 heroku[worker.1]: Process exited with status 1
2022-01-10T15:29:09.062362+00:00 heroku[worker.1]: State changed from up to crashed
2022-01-10T15:29:08.848414+00:00 app[worker.1]: npm ERR! Missing script: "start"
2022-01-10T15:29:08.848594+00:00 app[worker.1]: npm ERR! 
2022-01-10T15:29:08.848702+00:00 app[worker.1]: npm ERR! Did you mean one of these?
2022-01-10T15:29:08.848763+00:00 app[worker.1]: npm ERR!     npm star # Mark your favorite packages
2022-01-10T15:29:08.848909+00:00 app[worker.1]: npm ERR!     npm stars # View packages marked as favorites
2022-01-10T15:29:08.848946+00:00 app[worker.1]: npm ERR! 
2022-01-10T15:29:08.848978+00:00 app[worker.1]: npm ERR! To see a list of scripts, run:
2022-01-10T15:29:08.849071+00:00 app[worker.1]: npm ERR!   npm run
2022-01-10T15:29:08.850119+00:00 app[worker.1]: 
2022-01-10T15:29:08.850319+00:00 app[worker.1]: npm ERR! A complete log of this run can be found in:
2022-01-10T15:29:08.850361+00:00 app[worker.1]: npm ERR!     /app/.npm/_logs/2022-01-10T15_29_08_718Z-debug-0.log

**Comment by egorprnn** [comment, marked as answer]
Необходимо склонировать репозиторий из корня, не используйте сборку из релизов для Heroku. Также проверьте, что файл package.json загружается на Heroku. Скрипт его полностью поддерживает. 

UPD: Закрываю топик из-за неактивности, если будут вопросы - пингуйте тут же

**Comment by mrgulka** [comment]
При загрузке через консоль на хироку 3.0.1 релиза возникает это:
 git push heroku master -f
Enumerating objects: 34, done.
Counting objects: 100% (34/34), done.
Delta compression using up to 4 threads
Compressing objects: 100% (32/32), done.
Writing objects: 100% (34/34), 65.54 KiB | 2.26 MiB/s, done.
Total 34 (delta 0), reused 0 (delta 0), pack-reused 0
remote: Compressing source files... done.
remote: Building source:
remote:
remote: -----> Building on the Heroku-20 stack
remote: -----> Using buildpack: heroku/nodejs
remote: -----> Node.js app detected
remote:
remote: -----> Creating runtime environment
remote:
remote:        NPM_CONFIG_LOGLEVEL=error
remote:        NODE_VERBOSE=false
remote:        NODE_ENV=production
remote:        NODE_MODULES_CACHE=true
remote:
remote: -----> Installing binaries
remote:        engines.node (package.json):  >=17.0.0
remote:        engines.npm (package.json):   unspecified (use default)
remote:
remote:        Resolving node version >=17.0.0...
remote:        Downloading and installing node 18.0.0...
remote:        Using default npm version: 8.6.0
remote:
remote: -----> Restoring cache
remote:        - node_modules
remote:
remote: -----> Installing dependencies
remote:        Installing node modules
remote:        npm ERR! code ERESOLVE
remote:        npm ERR! ERESOLVE could not resolve
remote:        npm ERR!
remote:        npm ERR! While resolving: @typescript-eslint/eslint-plugin@4.33.0
remote:        npm ERR! Found: eslint@8.2.0
remote:        npm ERR! node_modules/eslint
remote:        npm ERR!   dev eslint@"^8.2.0" from the root project
remote:        npm ERR!   peer eslint@"*" from @typescript-eslint/experimental-utils@4.33.0
remote:        npm ERR!   node_modules/@typescript-eslint/experimental-utils
remote:        npm ERR!     @typescript-eslint/experimental-utils@"4.33.0" from @typescript-eslint/eslint-plugin@4.33.0
remote:        npm ERR!     node_modules/@typescript-eslint/eslint-plugin
remote:        npm ERR!       dev @typescript-eslint/eslint-plugin@"^4.22.1" from the root project
remote:        npm ERR!   1 more (eslint-utils)
remote:        npm ERR!
remote:        npm ERR! Could not resolve dependency:
remote:        npm ERR! peer eslint@"^5.0.0 || ^6.0.0 || ^7.0.0" from @typescript-eslint/eslint-plugin@4.33.0
remote:        npm ERR! node_modules/@typescript-eslint/eslint-plugin
remote:        npm ERR!   dev @typescript-eslint/eslint-plugin@"^4.22.1" from the root project
remote:        npm ERR!
remote:        npm ERR! Conflicting peer dependency: eslint@7.32.0
remote:        npm ERR! node_modules/eslint
remote:        npm ERR!   peer eslint@"^5.0.0 || ^6.0.0 || ^7.0.0" from @typescript-eslint/eslint-plugin@4.33.0
remote:        npm ERR!   node_modules/@typescript-eslint/eslint-plugin
remote:        npm ERR!     dev @typescript-eslint/eslint-plugin@"^4.22.1" from the root project
remote:        npm ERR!
remote:        npm ERR! Fix the upstream dependency conflict, or retry
remote:        npm ERR! this command with --force, or --legacy-peer-deps
remote:        npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
remote:        npm ERR!
remote:        npm ERR! See /tmp/npmcache.DrPJl/eresolve-report.txt for a full report.
remote:
remote:        npm ERR! A complete log of this run can be found in:
remote:        npm ERR!     /tmp/npmcache.DrPJl/_logs/2022-04-30T15_29_19_878Z-debug-0.log
remote:
remote: -----> Build failed
remote:
remote:        We're sorry this build is failing! You can troubleshoot common issues here:
remote:        https://devcenter.heroku.com/articles/troubleshooting-node-deploys
remote:
remote:        Some possible problems:
remote:
remote:        - Dangerous semver range (>) in engines.node
remote:          https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version
remote:
remote:        Love,
remote:        Heroku
remote:
remote:  !     Push rejected, failed to compile Node.js app.

**Reply by mrgulka** [reply]
@MrZillaGold

**Reply by egorprnn** [reply]
[Используйте сборку из master ветки](https://github.com/MrZillaGold/VK2Discord/archive/refs/heads/master.zip). В релизе 3.0.1 ошибка.

**Reply by mrgulka** [reply]
всё работает ❤спасибо :)
